config/default.json

```diff
@@ -7,6 +7,10 @@
         "ConnectionSecurity": "",
         "TLSCertFile": "",
         "TLSKeyFile": "",
+        "TLSMinVer": "1.2",
```

Comment:
I'm curious what value these 4 new settings add vs making them hard-coded?

Reply:
Because the entire point of this PR is to allow the selection of these things.

Reply:
Thanks Chris - I only ready MM-10861 but after reading https://mattermost.atlassian.net/browse/MM-10418, makes more sense 👍

Reply:
So basically it does a few things:
- Allow the admin to choose what's the TLS min version. While TLS 1.0 and 1.1 is considered insecure, and we made 1.2 default, there are sometimes sadly cases where you still might require the outdated versions. We set the secure version as default and allow an override to cater legacy users. Important to notice is that this will only impact users that use the built-in server directly, and not something like NGINX in front.
- Secure Transport forces the browser to request all resources via HTTPS: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
- Allowing the user to override ciphers may be required either to cater legacy clients which don't support modern ciphers, or to limit the number of ciphers even more. The new default preset sets it to currently considered secure ciphers, and allows overwrites in the edge case. Also only affects people who don't use a reverse proxy in front.